### PR TITLE
Introduce an alias for DQMEDAnalyzer

### PR DESCRIPTION
**Pull request by schneiml**
This branch modifies all DQM-related config files to use an alias `DQMEDAnalyzer` instead of `cms.EDAnalyzer` to declare their `DQMEDAnalyzer`s. This is a first step to changing the base of `DQMEDAnalyzer` (likely to `EDProducer` or `edm::one::EDProducer`), which can then be used to run DQM unscheduled, improve concurrency and reduce memory consumption.

Right now, this is a no-op and should not do anything, since the alias still points to `cms.EDAnalyzer` and the base class is not changed. The final conversion likely has to wait for an additional feature in the framework (thanks @Dr15Jones !).

Caveats:
- These changes were obviously done in an automated way, and not fully validated by hand.
- The list of `DQMEDAnalyzer`s came form [cmssdt DXR](https://cmssdt.cern.ch/dxr/CMSSW_9_4_0/search?q=%2Bderived%3ADQMEDAnalyzer), which is CMSSW_9_4. I might have missed newer modules.
- This PR will very quickly conflict with others. I can redo the changes fairly easily in that case, but this is the motivation to separate it from the functional changes.
- This PR touches all `DQMEDAnalyzer`s, including those that are not typical online/offline/relval DQM. In case you use them for other purposes, please contact core DQM that we _know_ about you. Otherwise, we will likely break your application going ahead. 

This is related to issue #15866. It is similar to #18751 in terms of changes.

### CODE REVIEW

**Comment by ghellwig** [comment]
please test

**Comment by dmitrijus** [comment]
+1

File: PhysicsTools/Heppy/test/crab/heppy_config.py [hunk 1]
```diff
@@ -1 +0,0 @@
-../example_autofill.py
```
Comment (schneiml) [inline PhysicsTools/Heppy/test/crab/heppy_config.py]: Note to self: No idea what happened there. This should be removed from the commit.

Comment (schneiml) [reply PhysicsTools/Heppy/test/crab/heppy_config.py]: Solution: was a symlink, turned into a copy due to the scripting. Will be fixed soon.

**Comment by dmitrijus** [comment]
@fabiocos Can this be merged as soon as possible? (But not in pre4, we don't really need it there.)

This is a simple configuration change which does affect a lot of packages.
The number of 'conflicting' pull requests will continue to grow over time, the earlier this goes in, the better.

**Comment by schneiml** [comment]
Fixed the Heppy symlink and force-pushed the fixed version.

**Comment by perrotta** [comment]
please test

**Comment by Martin-Grunewald** [comment]
please test

**Comment by ianna** [comment]
+1

**Comment by Martin-Grunewald** [comment]
Sorry, but why is that useful right now with a noop?
What prevents to wait and make the final change in one go to the eventually new final class, avoiding an alias?

**Comment by schneiml** [comment]
@Martin-Grunewald it is useful because it allows us to test and integrate future changes quicker, since they will only affect the core packages.

Handling a PR of this size is very painful, it takes forever to validate any change due to the huge number of checked out packages, and tests just fail "randomly". So I'd rather have this huge PR integrated where I have high confidence that it will not break anything (since it does not change anything), and have a more reliable development environment to work on the actual, functional changes (which are complicated enough on their own).

**Comment by Martin-Grunewald** [comment]
+1

**Comment by thomreis** [comment]
+1

**Comment by ghellwig** [comment]
+1

**Comment by kpedro88** [comment]
+1

**Comment by slava77** [comment]
+1

for https://github.com/cms-sw/cmssw/pull/21871	d14f997
- this is somewhat tangential to reco. The only file in the reco category  is RecoEgamma/Examples/python/electronAnalyzer_cfi.py 
- jenkins tests pass and comparisons show no differences as expected

**Comment by civanch** [comment]
+1

**Comment by dmitrijus** [comment]
+1

**Comment by fabiocos** [comment]
@perrozzi @efeyazgan could you please check and sign in case?

**Comment by efeyazgan** [comment]
+1

**Comment by fabiocos** [comment]
@Dr15Jones this is a massive code migration wrt to the old python syntax,  I would like your final confirmation you are ok with it before moving forward (possibly in next IB already)

**Comment by fabiocos** [comment]
+1

@Dr15Jones ok, thanks, I let this go tonight

**Pull request merged**